src/Plutus/Model/Validator.hs

```diff
@@ -104,6 +104,20 @@ instance HasValidator (TypedValidator datum redeemer) where
 instance HasAddress (TypedValidator datum redeemer) where
   toAddress = toAddress . toValidatorHash
 
+-- untyped validator
+
+instance HasDatum (Versioned Validator) where
```

Comment:
I would like to see the introduction of a newtype eg. `UntypedValidator` so that user doesn't have to know that it has to be some versioned validator, and for the other types, too.